app/src/main/java/com/example/triptracker/view/HomeScreen.kt

```diff
@@ -162,19 +205,131 @@ fun convertPinListToString(pinList: List<String>): String {
   }
 }
 
+/**
+ * Displays the search bar at the top of the screen
+ *
+ * @param onBackClicked: Function to call when the back button is clicked
+ * @param onSearchActivated: Function to call when the search bar is activated
+ * @param viewModel: HomeViewModel to use for searching itineraries
+ * @param isNoResultFound: Boolean to indicate if no results were found
+ * @param navigation: Navigation object to use for navigation, i.e when an itinerary is clicked
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SearchBar(
+    onBackClicked: () -> Unit = {},
+    onSearchActivated: (Boolean) -> Unit,
+    viewModel: HomeViewModel = viewModel(),
+    isNoResultFound: Boolean = false,
+    navigation: Navigation
+) {
+
+  var searchText by remember { mutableStateOf("") }
+  val items = viewModel.filteredItineraryList.value ?: listOf()
+  val focusManager = LocalFocusManager.current // Get access to the focus manager
+
+  // If the search bar is active (in focus or contains text), we'll consider it active.
+  var isActive by remember(searchText) { mutableStateOf(searchText.isNotEmpty()) }
+
+  androidx.compose.material3.SearchBar(
```

Comment:
It would be cleaner to import only the SearchBar composable from Android Compose rather than the entire package. For example this can be achieved by updating the import statement to import androidx.compose.material3.SearchBar.